types/underscore/underscore-tests.ts

```diff
@@ -486,6 +486,28 @@ _.chain(anyValue)
     .find(i => i.findBooleanFunction())
     .value();
 
+// verify that inline partial objects can be provided without error to where and findWhere for a union type collection
+// where no types in the union share the same property names
```

Comment:
My company's solution ends up with a new error around `Partial<T>` where we're assigning a `string | number | string[] | undefined` (which will actually always be a `string` in context) to a `properties` property that expects a `string | undefined`, which I was originally thinking was ok but as I think about it more I suppose it's not really important for the types for any property to match up exactly.

I guess ideally TS would make sure that at least one of the types for each property matched the provided type for that property, but I'm not entirely sure how to make that happen. Doing something like `U extends Partial<T>` for the `properties` parameter doesn't seem to work.

To preserve property name checking but drop type checking for the type of each property, I could make something like a `PartalAny<T>` that applies `Partial` to `T` but also changes all of the types of all of the properties to `any`. Thoughts?

Also, I had thought that only inline declarations were required to include at least one property that is in the expected partial type, but that doesn't appear to be the case, so I should drop the "inline" from this comment. However, I still think that a lack of any expected property on a `properties` object is still a good error to surface since calling one of these functions with a `properties` object that can't possibly match anything isn't a useful thing to do.

Reply:
Too, I guess ideally any solution applied here should not involve an additional generic type argument and an `extends` check since ideally any change here would also be applied to the `Iteratee` type union, and it would be nice to not make that more complicated.

Reply:
After having given this even more thought, I think my rationale for checking property names actually probably also applies to checking types - if the actual type of the value that ends up being provided as search criteria does not match one of the possible types on the items being searched, then it's not a useful search criteria. Therefore, I think I'll leave this as-is and require that folks either use type guards or assertions to remove types from a type union that might not be useful as search criteria.